### ISSUE TITLE
Sold out events can't be accessed over the REST API without authentication

### ISSUE THREAD

**Issue by mnelson4** [eventespresso/event-espresso-core#527]
When an event is sold out, its status is changed from `publish`, `protected` or `private` to `sold_out`. 
Right now, the REST API only shows events with the status `publish` to unauthenticated users. So when an event becomes sold out, it's suddenly not viewable publicly over the REST API.

This would be fixed if we treated `sold_out` events the same way as `publish` events, but we don't know if they were previous published, protected or private. So in order to fix this, we'd probably have to:
1. When an event is sold out, store its previous post status somewhere (maybe we do that? I vaguely remember this issue coming up before)
2. When an event is sold out, don't change it's status to `sold_out`
3. Refactor EE to work in Drupal instead of WordPress 

Josh pointed out there were previous codebase tickets: https://events.codebasehq.com/projects/event-espresso/tickets/11392 and https://events.codebasehq.com/projects/event-espresso/tickets/11393

**Comment by sethshoultes** [comment]
Ha! Number 3 has been requested a few times over the years...

@mnelson4 how much time to implement option one?

**Comment by joshfeck** [comment]
I'll advise against adding a setting that allows for the site admin to choose what happens. An option would be confusing with imperceptible utility at best. 

Related to #278

**Comment by mnelson4** [comment]
@sethshoultes lemme see what, if anything, we're saving...

@joshfeck I think agree, but I'm not 100% I follow. Did I or someone else suggest that? That sounds different from the very vague approaches I've suggested, it seems to me

**Comment by joshfeck** [comment]
In this case, adding an option is the very vague approach because what would the options be?

I think either of the first two approaches you presented here could work fine. The first could be leaned on as a quick fix (store the previous post status and make use of that) with the idea that we'll eventually deprecate the sold_out post status.

**Comment by mnelson4** [comment]
Oh I see, Seth said "Also, what about adding a setting that allows the site admin to choose what happens?" Somehow I skimmed over that. Sorry!! (I try to avoid that, but obviously not well enough this time).

> I think either of the first two approaches you presented here could work fine. The first could be leaned on as a quick fix (store the previous post status and make use of that) with the idea that we'll eventually deprecate the sold_out post status.

aha, yes we do store "_previous_event_status" postmeta when an event is sold out...

I created an exploratory branch on https://github.com/eventespresso/event-espresso-core/pull/536. If that gets accepted basically as-is, it should just be another hour to add some unit tests. If that requires more discussion, the sky is the limit time-wise @sethshoultes
